add token burning (#7)
fleet-sdk/docs: docs/token-burning.md

# Burning Tokens

> Burning tokens on Ergo is done similarly to [minting](./transaction-building.md#step-4-2-mint-a-token). As to mint, you need to add an inexistent token to outputs; to burn, you must add tokens as inputs and omit the number of tokens you want to burn from outputs.

Reviewer: ```suggestion
Burning tokens on Ergo is similar to [minting](./transaction-building.md#step-4-2-mint-a-token). Minting occurs whenever a new token is added to a transaction's output. Burning occurs when a transaction has fewer outputs than inputs. Most transactions have outputs and inputs that are equal. 
```

> Having such an implicit burning mechanism makes it easy to accidentally burn tokens. So that Fleet only allows burning through the `TransactionBuilder.BurnTokens()` method or by explicitly allowing it through `TransactionBuilder.configure(x => x.allowTokenBurning(true))` if a manual approach is needed.

Reviewer: ```suggestion
To protect against accidently burning tokens, Fleet checks to make sure inputs and outputs are equal. To burn tokens, the `TransactionBuilder.BurnTokens()` method is used to declare the tokenId and amount to burn. 
```

Reviewer: Thought it was cleaner to include the code for `TransactionBuilder.BurnTokens()` method right after mentioning it, then that you change the initial configuration to remove this protection.

The `BurnTokens` method accepts a single object or an array of objects describing which and how many tokens to burn.

<!-- prettier-ignore -->
```ts
new TransactionBuilder(creationHeight)
  .from(inputs)
  .burnTokens({ // [!code focus]
    tokenId: sigUSDTokenId, // [!code focus]
    amount: "10" // the amount of tokens you want to burn // [!code focus]
  }) // [!code focus]
  .sendChangeTo(changeAddress)
  .build();

> ```

Reviewer: ```suggestion
If a manual approach is needed, burning can be explicitly allowed through a configuration option: `TransactionBuilder.configure(x => x.allowTokenBurning(true))`
```
